src/core/loop_sccdag_attributes/src/SCCDAGAttrs.cpp

```diff
@@ -521,6 +543,114 @@ std::set<InductionVariable *> SCCDAGAttrs::
   return containedIVs;
 }
 
+std::vector<Value *> SCCDAGAttrs::checkIfPeriodic(SCC *scc,
+                                                  LoopForestNode *loopNode) {
+  if (this->sccToLoopCarriedDependencies.find(scc)
+      == this->sccToLoopCarriedDependencies.end()) {
+    return {};
+  }
+
+  /*
+   * Currently only handles SCCs with two nodes.
+   */
+  if (scc->numberOfInstructions() != 2)
+    return {};
+
+  for (auto edge : this->sccToLoopCarriedDependencies.at(scc)) {
+
+    /*
+     * Only look for loop-carried data dependencies.
+     */
+    if (!edge->isLoopCarriedDependence() || edge->isControlDependence())
+      continue;
```

Comment:
this `continue` is necessary because there are other edges from/to the SCC that we want to skip. once we have a loop-carried data dependence, the other checks downstream will filter it further.